lib/jsonapi/plugs/query_parser.ex

```diff
@@ -255,4 +246,14 @@ defmodule JSONAPI.QueryParser do
 
     struct(struct, processed_map)
   end
+
+  defp normalize_fields(fields) do
+    Deprecation.warn(:query_parser_dash)
+
+    if JString.field_transformation() == :underscore do
+      fields
+    else
+      JString.underscore(fields)
+    end
```

Comment:
I'm still unsure what this means.  If `field_transformation` is `:underscore`, then return as it.  Otherwise, underscore the keys?

Reply:
Yeah, it's a tad confusing, but the gist is that if your API is using underscored keys there's no need to underscore them. If your API is using dashes or camel keys, underscore they keys so that we can more easily use them in Elixir.

Reply:
Ah ok.  That makes sense.  My initial thought was, "hey if you want underscore, let me make sure they are underscore".  But since this predates me, seems ok to leave as it was!